spray-json/jvm/src/main/scala/spray/json/ProductFormats.scala

```diff
@@ -74,7 +74,7 @@ trait ProductFormats extends ProductFormatsInstances {
         _.getName.drop("copy$default$".length).takeWhile(_ != '(').toInt)
       val fields = clazz.getDeclaredFields.filterNot { f =>
         import Modifier._
-        (f.getModifiers & (TRANSIENT | STATIC | 0x1000 /* SYNTHETIC*/ )) > 0
+        (f.getModifiers & (TRANSIENT | STATIC | 0x1000 /* SYNTHETIC*/ )) > 0 || f.getName.endsWith("$outer")
```

Comment:
As https://github.com/lampepfl/dotty/issues/14083 was fixed in Scala 3.1.2, if you update the Scala 3 version this OR is no longer needed.